### DISCUSSION TITLE
Unlocking chip via avrdude/serialupdi?

### DISCUSSION THREAD

**Discussion by robmaxtech** [SpenceKonde/megaTinyCore#771] [Q&A] [answered]
Hello, I'm using avrdude 7.0 with a serialUPDI programmer quite frequently and it works 100% fine and fast. But now I ran into following question: How can I unlock a chip (ATTiny212 in my case) via avrdude? I know it's possible with pymcuprog (with the parameter --chip-erase-locked-device) but I'd like to avoid the python stuff (which has caused other trouble already).
I searched now for hours but couldn't find something useful...
Thanks for any pointer and help!

**Comment by SpenceKonde** [comment, marked as answer]
We plan to add that feature to prog.py/SerialUPDI in the near future,

Python the cause of problems? No, AVRdude using C is the cause of most of
it's problems. All future tools from me for upload including hyperupdi
(which is expected to nearly double the upload speed to tinyAVRs)  and the
standalone  gadget will be loaded using simple tools written in python
(with a clothespin over my nose. I hate python, but for things that run on
a PC, it is far better than the alternatives) and you don't need over  a
dozen build environments and a compiler toolchain expert. ever wonder why
the simple problems, poor error messages, etc of avrdude never got fixed?
That's why.. We know the wrong way to write and upload tool and I will have
no part in writing another tool that way.

On Wed, Aug 17, 2022 at 12:31 PM robmaxtech ***@***.***>
wrote:

> Hello, I'm using avrdude 7.0 with a serialUPDI programmer quite frequently
> and it works 100% fine and fast. But now I ran into following question: How
> can I unlock a chip (ATTiny212 in my case) via avrdude? I know it's
> possible with pymcuprog (with the parameter --chip-erase-locked-device) but
> I'd like to avoid the python stuff (which has caused other trouble already).
> I searched now for hours but couldn't find something useful...
> Thanks for any pointer and help!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/megaTinyCore/discussions/771>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEWY2XUEPAEGNECEEPALVZUHWHANCNFSM562GARWQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***

**Reply by robmaxtech** [reply]
Thanks for the insightful reply; seems I have to embrace Python ;-) Haha, good to hear you hate it but at the same time see it as the best tool for this tasks; I'm absolutely convinced by your statement!
Great to hear you will add the feature, and also to hear about your future developments. You do an absolutely awesome work; I'm sooo grateful!

**Reply by robmaxtech** [reply]
Ok, I was able to detect the reason for my unlock issue; the chip was completely bricked seemingly (maybe by wrongly set fuses). With pymcuprog I'm able to program, lock and unlock now as expected.
